### ISSUE TITLE
(Breaking) Change in `register_templates_directory`

### ISSUE THREAD

**Issue by Nerixyz** [sunng87/handlebars-rust#549]
Upgrading from version [4.2.2 to 4.3.0](https://github.com/sunng87/handlebars-rust/compare/v4.2.2...v4.3.0), specifically because of #500 (580acbae000893ce539670b83bd5ea216b6f8ddf), _might_ result in `register_templates_directory` not working identically.

In v4.2.2, you could register files with an extension that contained multiple `.`'s:

```rust
handlebars.register_templates_directory(".hbs.html", "templates");
```

This would result in all files _ending in_ `.hbs.html` to be loaded. After upgrading, this breaks, as now, [`Path::extension`](https://doc.rust-lang.org/std/path/struct.Path.html#method.extension) is used to determine the extension:
https://github.com/sunng87/handlebars-rust/blob/74cf5a2c8204af86313c5e79d388fa643645d116/src/registry.rs#L311-L316

Thus, in the example above, no files would ever match, since [`Path::extension`](https://doc.rust-lang.org/std/path/struct.Path.html#method.extension) considers `html` as the extension.

[The documentation](https://docs.rs/handlebars/4.3.5/handlebars/struct.Handlebars.html#method.register_templates_directory) doesn't mention anything regarding the format of `extension`.

If my use-case is invalid, then the docs should at least mention that `extension` must never contain a `.` (excluding the first character), or that [`Path::extension`](https://doc.rust-lang.org/std/path/struct.Path.html#method.extension) will be used to determine the extension. Furthermore, the changelog could mention, that this _previously accepted but invalid_ use will no longer work.

**Comment by sunng87** [comment]
hello @Nerixyz , your use case should be accepted by handlebars registry. Unfortunately our tests did not cover this case. Let's treat it as a bug and I'm going to fix it in 4.3.x releases.

**Comment by sunng87** [comment]
@Nerixyz The fix is released in 4.3.6
